controllers/seedgen_controller.go

```diff
@@ -256,6 +257,33 @@ func (r *SeedGeneratorReconciler) currentAcmCrds(ctx context.Context) (acmCrdLis
 	return
 }
 
+func (r *SeedGeneratorReconciler) sanitizePullSecret(ctx context.Context) error {
+	sanitizedPullSecret := lcautils.NewPullSecretFromDockerConfig([]byte(common.PullSecretEmptyData))
+
+	// Sanitize cluster's pull-secret from sensitive data
+	if err := common.RetryOnConflictOrRetriable(retry.DefaultBackoff, func() error {
+		return r.Client.Update(ctx, sanitizedPullSecret)
+	}); err != nil {
+		return fmt.Errorf("failed to update pull-secret in seed cluster: %w", err)
+	}
+
+	deadlineCtx, deadlineCancel := context.WithTimeout(ctx, 10*time.Minute)
+	defer deadlineCancel()
+	_ = wait.PollUntilContextCancel(deadlineCtx, 30*time.Second, true, func(ctx context.Context) (done bool, err error) {
+		r.Log.Info("Waiting for MCO to override pull-secret file")
+		dockerConfigJSON, err := os.ReadFile(filepath.Join(common.Host, common.ImageRegistryAuthFile))
+		if err != nil {
+			r.Log.Info(fmt.Sprintf("Failed to read %s file with error %s, will retry",
+				common.ImageRegistryAuthFile, err))
+			return false, nil
+		}
+		r.Log.Info(fmt.Sprintf("%s data is %s", common.ImageRegistryAuthFile, strings.TrimSpace(string(dockerConfigJSON))))
+		return strings.TrimSpace(string(dockerConfigJSON)) == string(sanitizedPullSecret.Data[".dockerconfigjson"]), nil
```

Comment:
Consider using https://pkg.go.dev/github.com/stretchr/testify/require#JSONEqf instead of comparing strings